lib/Demangling/DemanglerAssert.h

```diff
@@ -22,30 +22,25 @@
 #include "swift/Demangling/Demangle.h"
 #include "swift/Demangling/NamespaceMacros.h"
 
-#if SWIFT_RUNTIME
+#if !defined(NDEBUG) && !SWIFT_RUNTIME
 
-// In the runtime, DEMANGLER_ASSERT() returns an error
+// Except in the runtime, DEMANGLER_ASSERT() works like assert()
```

Comment:
On review, this comment should be improved.

Reply:
Maybe it becomes more readible if we flip if and else branch?

```
#if SWIFT_RUNTIME || defined(NDEBUG)
```

Reply:
yes that's probably better.